pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,30 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// ControlPlaneVersionChange detects kubernetes version has changed between new and old KubeadminControlPlane objects.
+func ControlPlaneVersionChange(oldKCP *controlplanev1.KubeadmControlPlane, newKCP *controlplanev1.KubeadmControlPlane) bool {
+	if oldKCP != nil && newKCP != nil {
+		return oldKCP.Spec.Version != newKCP.Spec.Version
+	}
+	return false
+}
+
+// ControlPlaneReplicasChange detects whether replica count has changed between new and old KubeadminControlPlane objects.
+func ControlPlaneReplicasChange(oldKCP *controlplanev1.KubeadmControlPlane, newKCP *controlplanev1.KubeadmControlPlane) bool {
+	if oldKCP != nil && newKCP != nil {
+		return oldKCP.Spec.Replicas != newKCP.Spec.Replicas
+	}
+	return false
+}
+
+// WorkerNodeReplicasChange detects whether replica count has changed between new and old MachineDeployment objects.
+func WorkerNodeReplicasChange(oldMD *clusterv1.MachineDeployment, newMD *clusterv1.MachineDeployment) bool {
```

Comment:
I remember we had a previous discussion about returning an `int` instead of a `bool` to detect change. Did we decide on using a bool instead?

Reply:
Thanks for catching that. I think `int` would be more reusable, updating to return that.

Reply:
What's the rationale for an int? What makes an int re-usable? Where does this need to be re-used?